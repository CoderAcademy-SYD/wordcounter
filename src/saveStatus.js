export const SUCCESS = 201;
export const FAILURE = 400;
export const WAITING = 666;
export const IDLE = 0;
