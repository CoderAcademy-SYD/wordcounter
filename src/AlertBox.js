import React from 'react';
import { SUCCESS, FAILURE, WAITING, IDLE } from './saveStatus'

export default function AlertBox({ status }) {
  if (status === WAITING) {
    return (<div className='pv2'>Waiting...</div>);
  } else if (status === SUCCESS) {
    return (<div className='pv2'>Successful</div>);
  } else if (status === FAILURE) {
    return (<div className='pv2'>Failed &#x1F61E;</div>);
  } else {
    return (null)
  }
}
