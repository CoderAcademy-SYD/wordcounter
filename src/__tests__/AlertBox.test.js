import React from 'react';
import renderer from 'react-test-renderer';
import AlertBox from '../AlertBox';
import { SUCCESS, IDLE } from '../saveStatus';

describe("AlertBox", () => {
  it("Displays the status", () => {
    const counter = renderer.create(
      <AlertBox status={SUCCESS}/>
    );
    expect(counter.toJSON()).toMatchSnapshot();
  });
});

