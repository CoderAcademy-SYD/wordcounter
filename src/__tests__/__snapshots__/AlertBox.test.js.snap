// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox Displays the status 1`] = `
<div
  className="pv2"
>
  Successful
</div>
`;
